### PR TITLE
3036 by Inlead: Adjust ding_redia_variables module.

### PR DESCRIPTION
**Pull request by bozeak**
https://platform.dandigbib.org/issues/3036

### CODE REVIEW

File: modules/ding_variables/ding_variables.info [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name = Ding Variables
+description = Exposed custom variables as a JSON format.
+core = 7.x
+package = Ding!
+version = 7.x-3.0.5
```
Comment (cableman) [inline modules/ding_variables/ding_variables.info]: Normally we don't have version tags inside the custom module files.

File: modules/ding_variables/ding_variables.module [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+/**
+ * Provide variables to the json output.
+ *
+ * @return array
+ *   List of all variables to the output.
+ */
+function ding_variables_display() {
+  // TODO: Find a way to remove hardcode.
```
Comment (cableman) [inline modules/ding_variables/ding_variables.module]: Should this "todo" be fixed or removed?

File: modules/ding_variables/ding_variables.module [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    'ting_search_profile',
+    'ting_search_results_per_page',
+    'ting_search_result_message_limit',
+    'ting_search_url',
+    'ting_sort_default',
+    'ting_ranking_custom',
+    'ting_ranking_fields',
+    'ting_boost_fields',
+    'ting_agency',
+  );
```
Comment (cableman) [inline modules/ding_variables/ding_variables.module]: Please note that within the 4.3.x the SAL release these variables have been moved into the opensearch module and I think most of them simply can be renamed to opensearch_*.

@kasperg can all these simply be renamed?

Comment (kasperg) [reply modules/ding_variables/ding_variables.module]: Yes. Check [`opensearch_migrate_settings()`](https://github.com/ding2/ding2/blob/master/modules/opensearch/opensearch.install#L18) for new names.

One notable exception is `ting_agency` - [that is bound to change later on though](https://platform.dandigbib.org/issues/3322).

File: modules/ding_variables/ding_variables.module [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+ * @return array
+ *   List of carousels pane configurations.
+ */
+function _ding_variables_get_carousel_configs($carousel) {
+  $query = db_select('panels_pane', 'pp');
+  $result = $query
+    ->fields('pp', array('subtype', 'configuration'))
+    ->condition('subtype', $carousel)
+    ->execute()
+    ->fetchAll();
```
Comment (kasperg) [inline modules/ding_variables/ding_variables.module]: This seems to assume that #1007 is merged. Is that true?

File: modules/ding_variables/ding_variables.info (hunk 1 above)
Comment (kasperg) [reply modules/ding_variables/ding_variables.info]: Also there are no dependencies mentioned here. That does not seem right.

File: ding2.install [hunk 5]
```diff
@@ -564,3 +564,10 @@ function ding2_update_7045() {
 function ding2_update_7046() {
   ding2_translation_update();
 }
+
+/**
+ * Enable ding_variables module.
+ */
+function ding2_update_7047() {
+  module_enable(array('ding_variables'));
```
Comment (kasperg) [inline ding2.install]: This probably only needs to be enabled for libraries using `opensearch` - especially if `opensearch` is added as a dependency.

File: modules/ding_variables/ding_variables.module (hunk 4 above)
Comment (bozeak) [reply modules/ding_variables/ding_variables.module]: As I know we are still waiting for this to be merged.

Comment (cableman) [reply modules/ding_variables/ding_variables.module]: An that PR is firste set to next release and therefor we can't use this before that. I have contact DDB in regards to this issue.

Comment (guddo) [reply modules/ding_variables/ding_variables.module]: Ok, let us know if we need to refactor to current release.

Comment (cableman) [reply modules/ding_variables/ding_variables.module]: Will do, awaiting Gitte and Rolf

Comment (cableman) [reply modules/ding_variables/ding_variables.module]: @guddo after talk with Gitte: We would like the carousels export to use the current data structure as the changes to the workflow (moving them into panes) is not set for this release. But the app has to be released before that release. 

Short version: Yes, please refactor.

**Comment by cableman** [comment]
Reviewed and awaiting release

**Pull request merged**